storage/mapTimeCache/mapTimeCache.go

```diff
@@ -0,0 +1,236 @@
+	ctx, mtc.cancelFunc = context.WithCancel(context.Background())
+
+	go func(ctx context.Context) {
+		timer := time.NewTimer(mtc.cacheExpiry)
+		defer timer.Stop()
+
+		for {
+			timer.Reset(mtc.cacheExpiry)
+
+			select {
+			case <-timer.C:
+				mtc.timeCache.Sweep()
+			case <-ctx.Done():
+				ltcLog.Info("closing sweep go routine...")
+				return
+			}
+		}
+	}(ctx)
+}
+
+// OnSweep is the handler called on Sweep method
+func (mtc *mapTimeCacher) OnSweep(key []byte) {
+	if key == nil {
+		return
+	}
+
+	mtc.Lock()
+	defer mtc.Unlock()
+
+	delete(mtc.dataMap, string(key))
+}
+
+// Clear deletes all stored data
+func (mtc *mapTimeCacher) Clear() {
+	mtc.Lock()
+	defer mtc.Unlock()
+
+	mtc.dataMap = make(map[string]interface{})
+}
+
+// Put adds a value to the cache. Returns true if an eviction occurred
+func (mtc *mapTimeCacher) Put(key []byte, value interface{}, _ int) (evicted bool) {
+	mtc.Lock()
+	defer mtc.Unlock()
+
+	_, evicted = mtc.dataMap[string(key)]
+	mtc.dataMap[string(key)] = value
+	if evicted {
+		mtc.upsertToTimeCache(key)
+		return true
+	}
+
+	mtc.addToTimeCache(key)
+	return false
+}
+
+// Get returns a key's value from the cache
+func (mtc *mapTimeCacher) Get(key []byte) (value interface{}, ok bool) {
+	mtc.RLock()
+	defer mtc.RUnlock()
+
+	v, ok := mtc.dataMap[string(key)]
+	return v, ok
+}
+
+// Has checks if a key is in the cache
+func (mtc *mapTimeCacher) Has(key []byte) bool {
+	mtc.RLock()
+	defer mtc.RUnlock()
+
+	_, ok := mtc.dataMap[string(key)]
+	return ok
+}
+
+// Peek returns a key's value from the cache
+func (mtc *mapTimeCacher) Peek(key []byte) (value interface{}, ok bool) {
+	mtc.RLock()
+	defer mtc.RUnlock()
+
+	v, ok := mtc.dataMap[string(key)]
+	return v, ok
+}
+
+// HasOrAdd checks if a key is in the cache.
+// If key exists, does not update the value. Otherwise, adds the key-value in the cache
+func (mtc *mapTimeCacher) HasOrAdd(key []byte, value interface{}, _ int) (has, added bool) {
+	mtc.Lock()
+	defer mtc.Unlock()
+
+	_, ok := mtc.dataMap[string(key)]
+	if ok {
+		return true, false
+	}
+
+	mtc.dataMap[string(key)] = value
+	mtc.upsertToTimeCache(key)
+
+	return false, true
+}
+
+// Remove removes the key from cache
+func (mtc *mapTimeCacher) Remove(key []byte) {
+	mtc.Lock()
+	defer mtc.Unlock()
+
+	delete(mtc.dataMap, string(key))
+}
+
+// Keys returns all keys from cache
+func (mtc *mapTimeCacher) Keys() [][]byte {
+	mtc.RLock()
+	defer mtc.RUnlock()
+
+	keys := make([][]byte, len(mtc.dataMap))
+	idx := 0
+	for k := range mtc.dataMap {
+		keys[idx] = []byte(k)
+		idx++
+	}
+	return keys
+}
+
+// Len returns the size of the cache
+func (mtc *mapTimeCacher) Len() int {
+	mtc.RLock()
+	defer mtc.RUnlock()
+
+	return len(mtc.dataMap)
+}
+
+// SizeInBytesContained returns the size in bytes of all contained elements
+func (mtc *mapTimeCacher) SizeInBytesContained() uint64 {
+	mtc.RLock()
+	defer mtc.RUnlock()
+
+	totalSize := 0
+	b := new(bytes.Buffer)
+	for _, v := range mtc.dataMap {
```

Comment:
I would simply keep a value for this functionality, adding to it or subtracting from it whenever we do an add or remove